src/manifest.c

```diff
@@ -94,6 +96,50 @@ static gboolean parse_image(GKeyFile *key_file, const gchar *group, RaucImage **
 	return res;
 }
 
+static gboolean parse_meta(GKeyFile *key_file, const gchar *group, RaucManifest *raucm, GError **error)
+{
+	g_auto(GStrv) groupsplit = NULL;
+	const gchar *groupname = NULL;
+	g_auto(GStrv) keys = NULL;
+	GError *ierror = NULL;
+
+	g_return_val_if_fail(key_file != NULL, FALSE);
+	g_return_val_if_fail(group != NULL, FALSE);
+	g_return_val_if_fail(raucm != NULL, FALSE);
+	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
+	groupsplit = g_strsplit(group, ".", 2);
+	if ((g_strv_length(groupsplit) != 2) || strchr(groupsplit[1], '.')) {
+		g_set_error(error, G_KEY_FILE_ERROR, G_KEY_FILE_ERROR_PARSE,
+				"invalid metadata section name '%s' (must contain a single '.')", group);
+		return FALSE;
+	}
+
+	groupname = g_intern_string(groupsplit[1]);
```

Comment:
What is the purpose of using interned strings here?

To my understanding, the primary use of interned strings is for simplifying string comparison which I do not see apply here.

Is this to omit manual string allocation and freeing?
Since interned strings are allocated in global scope, the allocation overhead should increase with each bundle being opened during the process lifetime.

Reply:
Mh, I don't see how this is different from all other places in the code where strings are handled.

I'd suggest adding at least a note in the code that intern strings are used for not having to write a free function for `RManifestMetaEntry`.

Reply:
I've changed it to use dynamic allocations for the strings.